CMakeLists.txt

```diff
@@ -285,7 +285,8 @@ else()
   set(DUCKDB_DEV_ITERATION 0)
   set(DUCKDB_VERSION "v${DUCKDB_MAJOR_VERSION}.${DUCKDB_MINOR_VERSION}.${DUCKDB_PATCH_VERSION}-dev${DUCKDB_DEV_ITERATION}")
 endif()
-
+add_definitions(-DDUCKDB_SOURCE_ID="\""${GIT_COMMIT_HASH}"\"")
```

Comment:
This has been problematic in the past because it means we always need to recompile the entire system whenever a change is made to the git hash, which happens after e.g. a `git commit`.